superset/views/core.py

```diff
@@ -2406,8 +2408,8 @@ def sql_json(self):
             status=QueryStatus.PENDING if async_ else QueryStatus.RUNNING,
             sql_editor_id=request.form.get('sql_editor_id'),
             tmp_table_name=tmp_table_name,
-            user_id=int(g.user.get_id()),
-            client_id=request.form.get('client_id'),
+            user_id=int(g.user.get_id()) if g.user and g.user.get_id() else None,
```

Comment:
I know that right now `admin` has user id 1, but in the future it could change to be 0 and it would break this code. I think it's better to be more explicit here an unravel the one liner:

```python
client_id = request.form.get('client_id') or utils.shortid()

try:
    user_id = int(g.user.get_id())
except Exception:
    user_id = None

query = Query(
    ...
    user_id=user_id,
    client_id=client_id,
)
```

Reply:
What I'm fixing here is that `g.user.get_id()` returns `None` when using the token auth mechanism on our fork. int(None) raises.

I can change this to `g.user.get_id() is not None`

Reply:
With the try block, `int(None)` would raise and `user_id` would be set to `None`.

Reply:
I'm not even sure that the int casting is necessary here, let me check

Reply:
Turns out it isn't necessary as far as I understand.